modules/inventory/server.lua

```diff
@@ -1533,13 +1533,13 @@ RegisterServerEvent('ox_inventory:updateWeapon', function(action, value, slot)
 				Inventory.RemoveItem(inventory, weapon.name, 1, weapon.metadata, weapon.slot)
 			elseif action == 'component' then
 				if type == 'number' then
-					Inventory.AddItem(inventory, weapon.metadata.components[value], 1)
+					Inventory.AddItem(inventory, weapon.metadata.components[value].name, 1)
 					table.remove(weapon.metadata.components, value)
 					weapon.weight = Inventory.SlotWeight(item, weapon)
 				elseif type == 'string' then
 					local component = inventory.items[tonumber(value)]
 					Inventory.RemoveItem(inventory, component.name, 1)
-					table.insert(weapon.metadata.components, component.name)
+					weapon.metadata.components = data
```

Comment:
Don't just take data from the client and update components with that; should be inserting like it was before.

Reply:
Not sure how this would be done other then getting the data from client, i could add a second check server side to validate all the data passed but other then that i would mean changing the logic of adding components from client to server.

Would data validation be enough ?

Reply:
I added a server side validation that is basically recreating the component table.
I initially wanted to used inventory.weapon and call ``DoesWeaponTakeWeaponComponent`` / ``HasPedGotWeaponComponent`` but those arent rpc.